checkpoint/side_handler.go

```diff
@@ -32,6 +34,39 @@ func NewSideTxHandler(k Keeper, contractCaller helper.IContractCaller) hmTypes.S
 	}
 }
 
+// SideHandleMsgCheckpointAdjust handles MsgCheckpointAdjust message for external call
+func SideHandleMsgCheckpointAdjust(ctx sdk.Context, k Keeper, msg types.MsgCheckpointAdjust, contractCaller helper.IContractCaller) (result abci.ResponseDeliverSideTx) {
+	logger := k.Logger(ctx)
+	chainParams := k.ck.GetParams(ctx).ChainParams
+	params := k.GetParams(ctx)
+
+	checkpointBuffer, err := k.GetCheckpointFromBuffer(ctx)
+	if checkpointBuffer != nil {
+		logger.Error("checkpoint buffer", "error", err)
+		return common.ErrorSideTx(k.Codespace(), common.CodeCheckpointBuffer)
+	}
+
+	rootChainInstance, err := contractCaller.GetRootChainInstance(chainParams.RootChainAddress.EthAddress())
+	if err != nil {
+		logger.Error("Unable to fetch rootchain contract instance", "error", err)
+		return common.ErrorSideTx(k.Codespace(), common.CodeOldCheckpoint)
+	}
+
+	root, start, end, _, proposer, err := contractCaller.GetHeaderInfo(msg.HeaderIndex, rootChainInstance, params.ChildBlockInterval)
+	if err != nil {
+		logger.Error("Unable to fetch checkpoint from rootchain", "error", err, "checkpointNumber", msg.HeaderIndex)
+		return common.ErrorSideTx(k.Codespace(), common.CodeNoCheckpoint)
+	}
+
+	if msg.EndBlock == end && msg.StartBlock == start && bytes.Equal(msg.RootHash.Bytes(), root.Bytes()) && bytes.Equal(msg.Proposer.Bytes(), proposer.Bytes()) {
+		logger.Error("Same Checkpoint in DB")
+		return common.ErrorSideTx(k.Codespace(), common.CodeCheckpointAlreadyExists)
+	}
```

Comment:
We want this to be the same. 


```suggestion
	if msg.EndBlock != end || msg.StartBlock != start || !bytes.Equal(msg.RootHash.Bytes(), root.Bytes()) && bytes.Equal(msg.Proposer.Bytes(), proposer.Bytes()) {
		logger.Error("Checkpoint on Rootchain is not same as msg")
		return common.ErrorSideTx(k.Codespace(), common.CodeCheckpointAlreadyExists)
	}
```

Reply:
ha ye sahi kr dia maine
testing ke waqt dekha